pyomo/core/base/var.py

```diff
@@ -82,24 +84,58 @@ def __init__(self, component=None):
     def has_lb(self):
         """Returns :const:`False` when the lower bound is
         :const:`None` or negative infinity"""
-        lb = self.lb
-        return (lb is not None) and \
-            (value(lb) != float('-inf'))
+        return self.lb not in _no_lower_bound
 
     def has_ub(self):
         """Returns :const:`False` when the upper bound is
         :const:`None` or positive infinity"""
-        ub = self.ub
-        return (ub is not None) and \
-            (value(ub) != float('inf'))
+        return self.ub not in _no_upper_bound
+
+    # TODO: deprecate this?  Properties are generally preferred over "set*()"
+    def setlb(self, val):
+        """
+        Set the lower bound for this variable after validating that
+        the value is fixed (or None).
+        """
+        self.lower = val
+
+    # TODO: deprecate this?  Properties are generally preferred over "set*()"
+    def setub(self, val):
+        """
+        Set the upper bound for this variable after validating that
+        the value is fixed (or None).
+        """
+        self.upper = val
 
     @property
     def bounds(self):
-        """Returns the tuple (lower bound, upper bound)."""
+        """Returns the tuple (lower bound, upper bound).
+
+        This returns the current (numeric) values of the lower and upper
+        bounds as a tuple.  If there is no bound, returns None (and not
+        +/-inf)
+
+        """
         return (self.lb, self.ub)
     @bounds.setter
     def bounds(self, val):
-        raise AttributeError("Assignment not allowed. Use the setub and setlb methods")
+        self.lower, self.upper = val
+
+    @property
+    def lb(self):
+        """Return the numeric value of the variable lower bound."""
+        return value(self.lower)
+    @lb.setter
+    def lb(self, val):
```

Comment:
If tests for `lb.setter` and `ub.setter` are relatively simple to write, we may want to add them.

Reply:
Also, I think that the docstrings for `_VarData` and `_GeneralVarData` needs to be updated now that these *do* have setters.